x-pack/plugins/apm/server/routes/services.ts

```diff
@@ -316,23 +318,60 @@ export const serviceThroughputRoute = createRoute({
       t.type({ transactionType: t.string }),
       uiFiltersRt,
       rangeRt,
+      comparisonRangeRt,
     ]),
   }),
   options: { tags: ['access:apm'] },
   handler: async ({ context, request }) => {
     const setup = await setupRequest(context, request);
     const { serviceName } = context.params.path;
-    const { transactionType } = context.params.query;
+    const {
+      transactionType,
+      comparisonStart,
+      comparisonEnd,
+    } = context.params.query;
     const searchAggregatedTransactions = await getSearchAggregatedTransactions(
       setup
     );
 
-    return getThroughput({
+    const { start, end } = setup;
+
+    const commonProps = {
       searchAggregatedTransactions,
       serviceName,
       setup,
       transactionType,
+    };
+
+    const throughputPromise = getThroughput({
+      ...commonProps,
+      start,
+      end,
     });
+
+    const comparisonPromise =
+      comparisonStart && comparisonEnd
+        ? getThroughput({
+            ...commonProps,
+            start: comparisonStart,
+            end: comparisonEnd,
+          })
+        : undefined;
+
+    const [currentPeriod, previousPeriod] = await Promise.all([
+      throughputPromise,
+      comparisonPromise,
```

Comment:
They are both for throughput, right?

```suggestion
      currentThroughputPeriodPromise,
      previousThroughputPeriodPromise,
```